src/main.rs

```diff
@@ -425,8 +430,73 @@ fn build_snapshot_settings_window(parent: &gtk::Window) {
 }
 
 // Take a snapshot of the current image and write it to the configured location
-fn take_snapshot() {
-    println!("tadam!");
+fn take_snapshot(pipeline: &gst::Pipeline) {
+    let settings = load_settings();
+
+    // Create the GStreamer caps for the output format
+    let (caps, extension) = match settings.format {
+        OutputFormat::JPEG => (gst::Caps::new_simple("image/jpeg", &[]), "jpg"),
+        OutputFormat::PNG => (gst::Caps::new_simple("image/png", &[]), "png"),
+    };
+
+    let sink = pipeline.get_by_name("sink").expect("sink not found");
+    let last_sample = sink.get_property("last-sample").unwrap();
+    let last_sample = match last_sample.get::<gst::Sample>() {
+        None => {
+            // We have no sample to store yet
```

Comment:
Maybe print an error or something?

Reply:
This can only happen if no frame was ever shown on the screen. Doesn't seem useful to print an error in that case if the user didn't see a frame yet anyway :)

Reply:
What happens if there is no input?

Reply:
We never write any snapshots, the user won't see any input. And at some point there would likely be an error from GStreamer